### DISCUSSION TITLE
macOS universal autofill

### DISCUSSION THREAD

**Discussion by memco** [kovidgoyal/kitty#8173] [Q&A]
I know that building password management into Kitty itself is not on the radar, but in recent versions of macOS many applications have an `Edit > AutoFill > Passwords...` menu item available that will trigger the UI for the built in Passwords app which could then be used to input a password: is it possible to access this feature with something like [menu_map](https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.menu_map) or could it be included as a future feature?

**Comment by kovidgoyal** [comment]
No clue, not an Apple user. In principle it can be done, but I am not sure
how. Issues you would need to solve:

1) Get some associated context for the password. kitty would need to
know what context the password is being asked for to pass it to the
system. Probably by scraping the screen? Check what the currently running
foreground process is? There are APIs in kitty to do either of those.
But the foreground process hack wont work when running over SSH, for
instance.

2) Write some cocoa code in kitty/cocoa_window.m to popup a dialog with
an NSSecureTextInput field (I am guessing from a quick google). Not
sure.

PR welcome!
